sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -434,24 +436,21 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
     left.dataType match {
       case StringType if right.foldable =>
         val sdf = classOf[SimpleDateFormat].getName
-        val fString = if (constFormat == null) null else constFormat.toString
-        val formatter = ctx.freshName("formatter")
-        if (fString == null) {
+        if (formatter == null) {
           ev.copy(code = s"""
```

Comment:
this can be: `ev.copy(code = "", isNull = "false", value = ctx.defaultValue(dataType))`